segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/IndexAppendProcessor.java

```diff
@@ -40,39 +40,52 @@ public class IndexAppendProcessor {
     static final Duration TIMEOUT = Duration.ofMinutes(1);
     private final ScheduledExecutorService indexSegmentUpdateExecutor;
     private final StreamSegmentStore store;
+    private final LatestItemSequentialProcessor<IndexAppend> latestItemProcessor;
 
     public IndexAppendProcessor(ScheduledExecutorService indexSegmentUpdateExecutor, StreamSegmentStore store) {
         this.indexSegmentUpdateExecutor = indexSegmentUpdateExecutor;
         this.store = store;
+        latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
     }
 
     /**
      * Appends index segment on a separate thread.
      * @param segmentName  segment name.
+     * @param indexSegmentEventSize  Index segment event size.
      */
-    protected void processAppend(String segmentName) {
+    protected void processAppend(String segmentName, long indexSegmentEventSize) {
         // Not updating index segment for transient and transaction type.
         if (isTransientSegment(segmentName) || isTransactionSegment(segmentName)) {
             return;
         }
-        LatestItemSequentialProcessor<String> latestItemProcessor = new LatestItemSequentialProcessor<>(this::handleIndexAppend, indexSegmentUpdateExecutor);
-        latestItemProcessor.updateItem(segmentName);
+        store.getStreamSegmentInfo(segmentName, TIMEOUT)
```

Comment:
For performance reasons we want this call to getInfo to be inside of handleIndexAppend.

Reply:
Moved to handleIndexAppend.